granian/_loops.py

```diff
@@ -19,7 +19,7 @@ def __contains__(self, key: str) -> bool:
     def keys(self) -> Iterable[str]:
         return self._data.keys()
 
-    def register(self, key: str) -> Callable[[], Callable[..., Any]]:
+    def register(self, key: str) -> Callable[[Callable[..., Any]], Callable[..., Any]]:
```

Comment:
Probably would be worth to define a custom type `Wrappable = Callable[..., Any]` and cleanup the code a bit

Reply:
Then I would prefer to use `WrappableT = TypeVar("WrappableT", bound=Callable[..., Any])` for more flexibility and type constraint. What is your opinion?

Reply:
👍 makes sense